_pages/install/integrations/opf-integration/opf-child-pages/1-open-payment-framework-in-spartacus.md

```diff
@@ -0,0 +1,267 @@
+;;OpfExplicitTermsAndConditionsComponent;OpfExplicitTermsAndConditionsComponent;OpfExplicitTermsAndConditionsComponent;false
+
+INSERT_UPDATE ContentSlot;$contentCV[unique=true];uid[unique=true];name;cmsComponents(uid, $contentCV)
+;;BodyContentSlot-checkoutOpfPaymentAndReview;Body Content Slot for Checkout OPF Payment And Review;CheckoutProgressComponent,CheckoutProgressMobileTopComponent,OpfCheckoutPaymentAndReviewComponent,CheckoutProgressMobileBottomComponent,OpfExplicitTermsAndConditionsComponent
+```
+
+### Configuring the OPF Payment Routing
+
+This configuration is particularly useful for integrating and tailoring payment verification workflows in an application that utilizes the Open Payment Framework (OPF) with Spartacus.
+
+By modifying the paths in this configuration, you can control how and where the application redirects users during specific payment verification scenarios.
+
+The provided code snippet modifies the routing configuration in a Spartacus application, defining custom routes for payment verification processes. Here is what can be configured:
+
+```ts
+provideConfig(<RoutingConfig>{
+  routing: {
+    routes: {
+      paymentVerificationResult: {
+        paths: ['opf/payment-verification-redirect/result'],
+      },
+      paymentVerificationCancel: {
+        paths: ['opf/payment-verification-redirect/cancel'],
+      },
+    },
+  },
+}),
+```
+
+#### Route Definitions
+
+**paymentVerificationResult**: Specifies the path to redirect the user after a successful payment verification. In this example, the path is `opf/payment-verification-redirect/result`.
+
+**paymentVerificationCancel**: Specifies the path to redirect the user if the payment verification is canceled. Here, the path is `opf/payment-verification-redirect/cancel`.
+
+#### Customizability
+
+Developers can adapt these routes to align with their application's URL structure, ensuring a seamless and coherent user navigation experience.
+
+### Configuring OPF Quick Buy
+
+Currently, the Quick Buy feature in OPF integration supports only ApplePay and GooglePay.
+
+By modifying this snippet, you can configure the integration of GooglePay as a payment provider for the Quick Buy feature in an application. Below are the configurable aspects:
+
+```ts
+provideConfig(<OpfQuickBuyConfig>{
+  providers: {
+    'googlePay': {
+      resourceUrl: 'https://pay.google.com/gp/p/js/pay.js',
+    } as OpfQuickBuyGooglePayProvider,
+  },
+}),
+```
+
+**resourceUrl** Specifies the external script or API endpoint required for the payment provider to function. This can be updated if the provider releases a new script version, changes its URL, or requires a custom endpoint for specific regions.
+
+You can configure additional payment providers by extending the providers object with their respective names and settings. For example, include ApplePay, PayPal, or other custom payment gateways.
```

Comment:
@FollowTheFlo @Matejk00 Should PayPal be mentioned in this final sentence if Quick Buy only supports Apple Pay and Google Pay, or am I misunderstanding what is being said here? I am almost wondering if this sentence is even necessary, and whether it can be simply deleted? 🙂

Reply:
I believe we can delete the whole paragraph:

"You can configure additional payment providers by extending the providers object with their respective names and settings. For example, include ApplePay, PayPal, or other custom payment gateways."

As at this moment only GooglePay is configurable in `OpfQuickBuyConfig`.

Reply:
Or re-phrase it to:

```
You can add configuration for another payment providers by extending the `providers` object with their respective names and settings.
```